cpp/src/parquet/metadata.h

```diff
@@ -182,6 +184,32 @@ class PARQUET_EXPORT ColumnChunkMetaData {
   std::unique_ptr<ColumnChunkMetaDataImpl> impl_;
 };
 
+// \brief DataPageStats stores statistics about a data page including number of
+// values and rows.
+class PARQUET_EXPORT DataPageStats {
+ public:
+  explicit DataPageStats(EncodedStatistics encoded_statistics, int32_t num_values,
+                             std::optional<int32_t> num_rows)
+      : encoded_statistics_(std::move(encoded_statistics)),
+        num_values_(num_values),
+        num_rows_(num_rows) {}
+
+  const EncodedStatistics& statistics() { return encoded_statistics_;}
+
+  int32_t num_values() const {
+    return num_values_;
+  }
+
+  std::optional<int32_t> num_rows() const {
+    return num_rows_;
+  }
+
+ private:
+  const EncodedStatistics encoded_statistics_;
```

Comment:
this is safe but it seems possible to maybe avoid the copy here, since stats need to use later?

Reply:
Done.